components/ai_chat/content/browser/associated_link_content.cc

```diff
@@ -0,0 +1,160 @@
+// Copyright (c) 2025 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this file,
+// You can obtain one at https://mozilla.org/MPL/2.0/.
+
+#include "brave/components/ai_chat/content/browser/associated_link_content.h"
+
+#include <utility>
+
+#include "base/functional/bind.h"
+#include "base/logging.h"
+#include "base/strings/string_number_conversions.h"
+#include "base/uuid.h"
+#include "brave/components/ai_chat/content/browser/page_content_fetcher.h"
+#include "brave/components/ai_chat/core/browser/associated_content_delegate.h"
+#include "content/public/browser/browser_context.h"
+#include "content/public/browser/navigation_handle.h"
+#include "content/public/browser/web_contents.h"
+#include "net/base/net_errors.h"
+#include "ui/base/page_transition_types.h"
+
+namespace ai_chat {
+
+AssociatedLinkContent::AssociatedLinkContent(
+    GURL url,
+    std::u16string title,
+    content::BrowserContext* browser_context) {
+  DVLOG(2) << __func__ << "Creating link content for: " << url.spec()
+           << " title: " << title;
+
+  set_uuid(base::Uuid::GenerateRandomV4().AsLowercaseString());
+  set_url(std::move(url));
+  SetTitle(std::move(title));
+
+  // Create background WebContents optimized for headless loading
+  content::WebContents::CreateParams params(browser_context);
+  params.initially_hidden = true;
+  params.preview_mode = true;
+  web_contents_ = content::WebContents::Create(params);
+
+  // Start observing the WebContents
+  content::WebContentsObserver::Observe(web_contents_.get());
+}
+
+AssociatedLinkContent::~AssociatedLinkContent() = default;
+
+void AssociatedLinkContent::GetContent(GetPageContentCallback callback) {
```

Comment:
This needs unit test coverage.

Reply:
addressed in https://github.com/brave/brave-core/pull/30963/commits/6a20bdc50e39ab2ca5cafdb9eaedee3b328c0895